compression.go

```diff
@@ -75,21 +75,21 @@ func newCompressionPool(
 	}
 }
 
-func (c *compressionPool) Decompress(dst *bytes.Buffer, src *bytes.Buffer, readMaxBytes int) *Error {
+func (c *compressionPool) Decompress(dst *bytes.Buffer, src *bytes.Buffer, readMaxBytes int64) *Error {
```

Comment:
compressionPool isn't an exported type so we can change this to avoid casting in the logic below.